src/utils.c

```diff
@@ -254,3 +257,95 @@ is_scanner_only_pref (const char *pref)
     return 1;
   return 0;
 }
+
+/** @brief Writes scripts stats into a file.
+ *
+ * @param buf String to write.
+ * @param scan_id Scan ID for the file name.
+ * @param mode 2 to create the file, 0 to append text to the file,
+ *             1 to finish the json list removing the trailing comma before
+ *             appending the last text in the buffer.
+ *
+ */
+void
+write_script_stats (const char *buf, const char *scan_id, int mode)
+{
+  char *p = NULL;
+  FILE *path = NULL;
+  if (!prefs_get ("report_scripts"))
+    return;
+  p =
+    g_strdup_printf ("%s/%s-stats.json", prefs_get ("report_scripts"), scan_id);
```

Comment:
What is the reason to not let the caller provide the full filename instead of limiting that to report_scripts? 

From my point of view this is a utility function that whatever is in buf into a file so it would be useful for other parts as well when a caller would have the possibility to provide the path instead of internally create the path in here.